controllers/vm_controller.go

```diff
@@ -58,7 +58,11 @@ func (r *VmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
 	vm := kubevirtv1.VirtualMachine{}
 	if err := r.client.Get(ctx, req.NamespacedName, &vm); err != nil {
 		if errors.IsNotFound(err) {
-			// VM was deleted, so we can ignore it
+			// VM was deleted
+			vm.Name = req.Name
```

Comment:
If the VM was deleted, why do we add it to the metrics? Also it looks error-prone to call `metrics.SetVmWithVolume` with nil-pointers, as IIUC this func does not check for nil pointers sufficiently.

Reply:
- the issue was that the controller kept exposing the metric that the VM, although already deleted, as the problematic volume configuration
this way when a VM is deleted, the controller no longer has a true value for the metric

- I think the nil check in the function covers this case completely:

```
if pv == nil || pvc == nil {
  vmRbdVolume.WithLabelValues(vm.Name, vm.Namespace).Set(0)
  return
}
```

Reply:
https://github.com/kubevirt/ssp-operator/blob/33b6ca71959e903fc1f7abf813ef38d110061a10/pkg/monitoring/metrics/rbd_metrics.go#L29

The function that is called looks different, than what you quoted?

Reply:
I also changed that function on this PR:
https://github.com/kubevirt/ssp-operator/pull/742/files#diff-387c26f005e40fae497f3a5c01da1f877fc4eee90da11b7d8cf47512d8474b82R29-R31

Reply:
Right, my mistake.